src/test/scala/chiselTests/BoringUtilsTapSpec.scala

```diff
@@ -470,13 +470,7 @@ class BoringUtilsTapSpec extends ChiselFlatSpec with ChiselRunners with Utils wi
 
     class Foo extends RawModule {
       val a = WireInit(DecoupledIO(Bool()), DontCare)
-      val dummyA = Wire(Output(chiselTypeOf(a)))
-      // FIXME we shouldn't need this intermediate wire
-      // https://github.com/chipsalliance/chisel/issues/3557
-      dummyA :#= a
-      dontTouch(a)
-
-      val bar = Module(new Bar(dummyA))
+      val bar = Module(new Bar(a))
```

Comment:
I'm a little confused by this test `Bar` is a child of `Foo` yet we're taping and reading a value from `Foo` in `Bar`?

Reply:
Yep, that's what is happening!

"Tap" kinda has come to mean "run probes up, non-probes down" and this is just that second half.  And the issue here is specifically when this is ONLY the second half (never a probe) we expect the "fully aligned" but nothing makes that happen so emit a busted connect.  The input port created is fully aligned, but since we didn't (and shouldn't matter if we did) have to probe something we need a way to connect the two.  And the secret commands are limited, here I'm just taking all of that as fixed and working within what we've allowed there.

This is kinda the intuitive backing for the approach taken here, not that it matters really but .. now we read+probe (at least if that does anything) even if didn't have to run probes upwards or probe at all, which unsurprisingly is effective for providing uniform behavior.
(not that I don't wish/think we could do something better but also it's maybe less random that it appears, lol)

Reply:
Got it, thanks for the explanation!